lib/twitter-ads/http/request.rb

```diff
@@ -71,8 +71,37 @@ def oauth_request
       token    = OAuth::AccessToken.new(consumer, @client.access_token, @client.access_token_secret)
       request.oauth!(consumer.http, consumer, token)
 
+      handle_rate_limit = @client.options.fetch(:handle_rate_limit, false)
+      retry_max         = @client.options.fetch(:retry_max, 0)
+      retry_delay       = @client.options.fetch(:retry_delay, 1500)
+      retry_on_status   = @client.options.fetch(:retry_on_status, [500, 503])
```

Comment:
Why does this only have 500 and 503 (just trying to understand what it's doing)

Reply:
I see that it's for making sure the retry depends upon client side or server side error - maybe we can have specific server_side_retry_status_codes or something to make it even clearer

Reply:
Uses can override this via options parameter.